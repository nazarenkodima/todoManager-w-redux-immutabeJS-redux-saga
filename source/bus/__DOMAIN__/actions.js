
//Actions
import { types } from './types';

export const action = {
    action: () => {
        return {
            type: types.TYPE,
        };
    },
}
;
