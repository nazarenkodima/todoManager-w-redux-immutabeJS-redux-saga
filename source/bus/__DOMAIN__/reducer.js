//Core
import { Map } from 'immutable';

import { types } from './types';

const initialState = Map({});

export const Reducer = (state = initialState, action) => {
    switch (action.type) {
        case types.TYPE:
            return state;

        default:
            return state;
    }
};
