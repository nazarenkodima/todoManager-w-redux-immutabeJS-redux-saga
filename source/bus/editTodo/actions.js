
//Actions
import { types } from './types';

export const editActions = {
    editTodo: (id, message) => {
        return {
            type:    types.EDIT_TODO,
            payload: { id, message },

        };
    },
    updateEditedTodo: (message) => {
        return {
            type:    types.UPDATE_EDITED_TODO,
            payload: message,
        };
    },
    editCancel: () => {
        return {
            type: types.EDIT_CANCEL,
        };
    },
};
