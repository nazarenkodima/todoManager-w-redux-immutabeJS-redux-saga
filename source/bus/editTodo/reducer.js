//Core
import { Map, fromJS } from 'immutable';

import { types } from './types';

const initialState = Map({
    id:         '',
    newMessage: '',
});

export const editTodoReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.EDIT_TODO:
            return state.merge(
                fromJS({
                    id:         action.payload.id,
                    newMessage: action.payload.message,
                })
            );

        case types.UPDATE_EDITED_TODO:
            return state.set('newMessage', action.payload);

        case types.EDIT_CANCEL:
            return initialState;

        default:
            return state;
    }
};
