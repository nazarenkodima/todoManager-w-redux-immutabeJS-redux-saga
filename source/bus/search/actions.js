
//Actions
import { types } from './types';

export const searchActions = {
    searchTodo: (text) => {
        return {
            type:    types.SEARCH_TODO,
            payload: text,
        };
    },
}
;
