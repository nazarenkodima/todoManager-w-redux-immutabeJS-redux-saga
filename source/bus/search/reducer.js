//Core
import { Map } from 'immutable';

import { types } from './types';

const initialState = Map({
    searchTodo: '',
});

export const searchReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.SEARCH_TODO:
            return state.set('searchTodo', action.payload);

        default:
            return state;
    }
};
