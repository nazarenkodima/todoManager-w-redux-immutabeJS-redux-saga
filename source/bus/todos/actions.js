
//Actions
import { types } from './types';

export const todoActions = {
    //Sync
    fillTodos: (todos) => {
        return {
            type:    types.FILL_TODOS,
            payload: todos,
        };
    },
    createTodo: (todo) => {
        return {
            type:    types.CREATE_TODO,
            payload: todo,
        };
    },
    removeTodo: (todoID) => {
        return {
            type:    types.REMOVE_TODO,
            payload: todoID,
        };
    },
    //Async
    fetchTodosAsync: () => {
        return {
            type: types.FETCH_TODOS_ASYNC,
        };
    },
    createTodoAsync: (todo) => {
        return {
            type:    types.CREATE_TODO_ASYNC,
            payload: todo,
        };
    },
    removeTodoAsync: (todoID) => {
        return {
            type:    types.REMOVE_TODO_ASYNC,
            payload: todoID,
        };
    },
}
;
