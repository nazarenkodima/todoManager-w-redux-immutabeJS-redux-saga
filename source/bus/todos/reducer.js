//Core
import { fromJS, List } from 'immutable';

//Types
import { types } from './types';

const initialState = List();

export const todosReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_TODOS:
            return fromJS(action.payload);

        case types.CREATE_TODO:
            return state.unshift(fromJS(action.payload));

        case types.REMOVE_TODO:
            return state.filter((todo) => todo.get('id') !== action.payload);

        case types.UPDATE_TODO:
            return state.update(state.findIndex((todo) => todo.get('id') === action.payload.id),
                (todo) => todo.set('completed', !todo.get('completed')));

        case types.COMPLETE_ALL_TODO:
            return state.map((todo) => todo.set('completed', true));

        default:
            return state;
    }
};
