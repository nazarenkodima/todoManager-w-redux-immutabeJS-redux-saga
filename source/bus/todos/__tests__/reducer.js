//Core
import { List, fromJS } from "immutable";

//Reducer
import { todosReducer } from "../reducer";

//Actions
import { todoActions } from "../actions";

const initialState = List();

describe("todoReducer:", () => {
    test("should return initial state by default", () => {
        expect(todosReducer(void 0, {})).toEqual(initialState);
    });

    test("should handle FILL_TODOS action", () => {
        expect(todosReducer(void 0, todoActions.fillTodos(__.todos))).toEqual(
            fromJS(__.todos)
        );
    });

    test("should handle CREATE_TODO action", () => {
        expect(todosReducer(void 0, todoActions.createTodo(__.todos[0])))
            .toMatchInlineSnapshot(`
Immutable.List [
  Immutable.Map {
    "completed": false,
    "favorite": false,
    "id": "5cdc0efeb808f56fcaf45470",
    "message": "hello",
  },
]
`);
    });

    test("should handle REMOVE_TODO action", () => {
        expect(
            todosReducer(fromJS(__.todos), todoActions.removeTodo(__.todos[0].id))
        ).toMatchSnapshot();
    });

    test("should handle UPDATE_TODO action", () => {
        expect(
            todosReducer(fromJS(__.todos), todoActions.updateTodo(__.todos[0]))
        ).toEqual(fromJS(__.todos));
    });

    test("should handle COMPLETE_ALL_TODO action", () => {
        expect(
            todosReducer(fromJS(__.todos), todoActions.completeAllTodo(__.todos[0]))
        ).toMatchInlineSnapshot(`
Immutable.List [
  Immutable.Map {
    "completed": true,
    "favorite": false,
    "id": "5cdc0efeb808f56fcaf45470",
    "message": "hello",
  },
]
`);
    });
});
