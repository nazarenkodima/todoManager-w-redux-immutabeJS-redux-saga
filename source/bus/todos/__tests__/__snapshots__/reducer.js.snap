// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoReducer: should handle REMOVE_TODO action 1`] = `Immutable.List []`;
